### PR TITLE
server: reduce memcopy when record the statistics

### PR DESCRIPTION
**Pull request by nolouch**
## What have you changed? (required)

```
     flat  flat%   sum%        cum   cum%
     0.04s  0.11%  0.16%     23.11s 63.04%  github.com/pingcap/pd/server.(*RaftCluster).HandleRegionHeartbeat
     0.13s  0.35%  0.52%     23.06s 62.90%  github.com/pingcap/pd/server.(*clusterInfo).handleRegionHeartbeat
     0.05s  0.14%  0.65%      8.96s 24.44%  github.com/pingcap/pd/vendor/github.com/gogo/protobuf/proto.Clone
     0.55s  1.50%  2.15%      8.21s 22.39%  github.com/pingcap/pd/vendor/github.com/gogo/protobuf/proto.mergeStruct
     0.13s  0.35%  2.51%      6.51s 17.76%  github.com/pingcap/pd/server/core.(*StoreInfo).Clone
     0.01s 0.027%  2.54%      6.35s 17.32%  github.com/pingcap/pd/server/core.(*RegionsInfo).SetRegion
     0.02s 0.055%  2.59%      6.32s 17.24%  github.com/pingcap/pd/server/core.(*StoresInfo).GetStore
         0     0%  2.59%      6.24s 17.02%  github.com/pingcap/pd/server/core.(*RegionsInfo).AddRegion
     0.05s  0.14%  2.73%      6.05s 16.50%  github.com/pingcap/pd/server.(*clusterInfo).getRegionStoresLocked
     0.03s 0.082%  2.81%      5.52s 15.06%  runtime.systemstack
     0.01s 0.027%  2.84%      5.22s 14.24%  github.com/pingcap/pd/server/core.(*regionTree).update
     0.64s  1.75%  4.58%      5.08s 13.86%  github.com/pingcap/pd/vendor/github.com/google/btree.(*node).iterate
     0.02s 0.055%  4.64%      4.78s 13.04%  github.com/pingcap/pd/vendor/github.com/sirupsen/logrus.Infof

```

 Reduce the store clone.
## What are the type of the changes (required)?

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested (required)?
Manual.

### CODE REVIEW

File: server/cluster_info.go [hunk 1]
```diff
@@ -367,9 +367,19 @@ func (c *clusterInfo) GetRegionStores(region *core.RegionInfo) []*core.StoreInfo
 }
 
 func (c *clusterInfo) getRegionStoresLocked(region *core.RegionInfo) []*core.StoreInfo {
-	var stores []*core.StoreInfo
-	for id := range region.GetStoreIds() {
-		if store := c.core.Stores.GetStore(id); store != nil {
+	stores := make([]*core.StoreInfo, 0, len(region.GetPeers()))
+	for _, p := range region.GetPeers() {
```
Comment (liukun4515) [inline server/cluster_info.go]: What is the difference between `region.GetStoreIds` and `region.GetPeers` to get the `storeid` ？

Comment (nolouch) [reply server/cluster_info.go]: `GetStoreIds` will make a map, the `GetPeers` do not alloc new memory.

**Comment by liukun4515** [comment]
LGTM

File: server/core/store.go [hunk 2]
```diff
@@ -348,7 +348,16 @@ func (s *StoresInfo) GetStore(storeID uint64) *StoreInfo {
 	return store.Clone()
 }
 
-// SetStore set a StoreInfo with storeID
+// TakeStore returns the point of the origin StoreInfo with the specifiled storeID.
```
[the remark this answers is not in the document]
Comment (siddontang) [reply server/core/store.go]: Em, I prefer GetStore returns the reference, Take here confuses me. Why not only provide GetStore and let user Clone by him/herself?

[the remark this answers is not in the document]
Comment (nolouch) [reply server/core/store.go]: we need finished the clone in Lock.

**Pull request merged**